crates/bevy_render/src/pipeline/vertex_buffer_descriptor.rs

```diff
@@ -9,12 +9,13 @@ pub struct VertexBufferDescriptor {
     pub name: Cow<'static, str>,
     pub stride: u64,
     pub step_mode: InputStepMode,
-    pub attributes: Vec<VertexAttributeDescriptor>,
+    pub attributes: Vec<VertexAttributeDescriptor>, //TODO: remove the vec? should it be ever possible to have multiple attributes per buffer?
```

Comment:
yeah this should be just a single VertexAttributeDescriptor now, and perhaps (regarding earlier comments), instead of relying on the VertexBufferDescriptor `name` field, we remove that and instead make `VertexAttributeDescriptor` impl `Hash` based on both its format and its own `name` field and then we can use the `VertexAttributeDescriptor` as the hashmap key above (or for now just remove the Vec and check based on the attribute name rather than the buffer name).